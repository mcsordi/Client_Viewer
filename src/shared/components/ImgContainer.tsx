import React from 'react';

type TPath = {
  path: string;
};

export const ImgContainer: React.FC<TPath> = ({ path }) => {
  return (
    <div className="flex w-full h-44 border-b-2 items-center justify-center">
      <img
        alt={'profile image'}
        className="w-24 h-24 rounded-full"
        src={path}
      />
    </div>
  );
};
