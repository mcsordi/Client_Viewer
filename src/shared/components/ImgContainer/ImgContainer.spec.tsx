import { render, screen } from '@testing-library/react';
import { ImgContainer } from './ImgContainer';

describe('<ImgContainer/>', () => {
  it('should render an img with a image adress', () => {
    const { container } = render(<ImgContainer path="image.png" />);
    expect(container.firstChild?.firstChild).toHaveAttribute(
      'src',
      'image.png',
    );
    expect(container.firstChild?.firstChild).toHaveAttribute(
      'alt',
      'profile image',
    );
  });
  it('should match a snapshot', () => {
    const { container } = render(<ImgContainer path="image.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
