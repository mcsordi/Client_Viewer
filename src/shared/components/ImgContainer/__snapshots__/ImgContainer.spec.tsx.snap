// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImgContainer/> should match a snapshot 1`] = `
<div
  class="flex w-full h-44 border-b-2 items-center justify-center"
>
  <img
    alt="profile image"
    class="w-24 h-24 rounded-full"
    src="image.png"
  />
</div>
`;
